src/diffusers/loaders.py

```diff
@@ -1018,24 +1040,94 @@ def lora_state_dict(
         else:
             state_dict = pretrained_model_name_or_path_or_dict
 
+        # TODO(SAYAK) - the following mapping is needed whenever LoRAs are published
+        # in original SGM format (this was also meant here: https://github.com/huggingface/diffusers/pull/4287#discussion_r1275044911)
+        if unet_config is not None:  # need better chec
+            # use unet config to remap block numbers
+            state_dict = cls._map_sgm_blocks_to_diffusers(state_dict, unet_config)
```

Comment:
Need to be able to detect when `unet_config` should be optional. I think it's only relevant for SDXL LoRAs.